src/duration.js

```diff
@@ -862,6 +869,16 @@ export default class Duration {
     return clone(this, { values: negated }, true);
   }
 
+  /**
+   * Removes all units with values equal to 0 from this Duration.
+   * @example Duration.fromObject({ years: 2, days: 0, hours: 0, minutes: 0 }).removeZeroes().toObject() //=> { years: 2 }
+   * @return {Duration}
+   */
+  removeZeroes() {
+    const vals = removeZeroes(this.toObject());
+    return clone(this, { values: vals }, true);
```

Comment:
Two things here:
1. Please handle invalid Durations here. `Duration#removeZeroes` should return an invalid Duration again when called on an invalid Duration. You can check `Duration#plus` for how to do that. Ideally, add a test case for this, too.
2. `removeZeroes` already clones the object, so you can pass in `this.values` directly into it instead of making another copy via `this.toObject()`.